controlplane/kubeadm/api/v1alpha4/kubeadm_control_plane_webhook.go

```diff
@@ -101,6 +105,10 @@ func (in *KubeadmControlPlane) ValidateUpdate(old runtime.Object) error {
 		{spec, kubeadmConfigSpec, clusterConfiguration, "dns", "imageRepository"},
 		{spec, kubeadmConfigSpec, clusterConfiguration, "dns", "imageTag"},
 		{spec, kubeadmConfigSpec, clusterConfiguration, "imageRepository"},
+		{spec, kubeadmConfigSpec, clusterConfiguration, apiServer, "*"},
```

Comment:
Just to check, this includes also `CertSANs`, and `TimeoutForControlPlane`, is this intentional?
(despite that I don't see any problem in allowing mutation of those fields)

Reply:
Yes, this was intentional, and the tests include them as well. However, I'm not fully aware if there might be any unintended consequences, happy to test any edge cases that you think might break things.